Homework-06-documentation-testsanswers.ipynb

```diff
@@ -0,0 +1,201 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# nothing to code here"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Assignment 2\n",
+    "\n",
+    "Write a test case for the `count_words` method in the following class."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "test__init (__main__.TestTextStats) ... ok\n",
+      "test_count_words (__main__.TestTextStats) ... ERROR\n",
+      "\n",
+      "======================================================================\n",
+      "ERROR: test_count_words (__main__.TestTextStats)\n",
+      "----------------------------------------------------------------------\n",
+      "Traceback (most recent call last):\n",
+      "  File \"<ipython-input-5-7a63d35ea9d3>\", line 28, in test_count_words\n",
+      "    stat=TextStats(text)\n",
+      "TypeError: __init__() takes 1 positional argument but 2 were given\n",
+      "\n",
+      "----------------------------------------------------------------------\n",
+      "Ran 2 tests in 0.004s\n",
+      "\n",
+      "FAILED (errors=1)\n"
+     ]
+    },
+    {
+     "ename": "TypeError",
+     "evalue": "__init__() takes 1 positional argument but 2 were given",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
+      "\u001b[0;32m<ipython-input-5-7a63d35ea9d3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     34\u001b[0m \u001b[0mtext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"Betty Botter bought some butter. But she said the butter’s bitter. If I put it in my batter, it will make my batter bitter. But a bit of better butter will make my batter better. So it was better Betty Botter bought a bit of better butter.\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     35\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 36\u001b[0;31m \u001b[0mstat\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextStats\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtext\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     37\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount_words\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
+      "\u001b[0;31mTypeError\u001b[0m: __init__() takes 1 positional argument but 2 were given"
+     ]
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "class TextStats:\n",
+    "    \n",
+    "    def __init__(self):\n",
```

Comment:
It looks like you changed the TextStats class here, why? The next line should throw an error since `text` is not defined.